apps/evans/components/core/EmbeddedAudio.tsx

```diff
@@ -1,60 +1,176 @@
-import React from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
 import cx from 'classnames';
-import { LANG } from '@/lib/env';
+import { BackwardIcon, ForwardIcon } from '@heroicons/react/24/solid';
+import styles from '../../styles/AudioPlayer.module.css';
+import { formatDuration } from '@/lib/dates';
 
 interface Props {
-  className?: string;
-  title: string;
-  trackId: number;
-  playlistId?: number | null;
-  showArtwork?: boolean;
-  height?: number;
+  tracks: Array<{
+    title: string;
+    url: string;
+  }>;
 }
 
-const EmbeddedAudio: React.FC<Props> = ({
-  className = ``,
-  trackId,
-  playlistId,
-  showArtwork = true,
-  height = 166,
-  title,
-}) => (
-  <iframe
-    className={cx(`bg-flgray-100`, className)}
-    title={title}
-    width="100%"
-    height={String(height)}
-    scrolling="no"
-    frameBorder="no"
-    src={src(trackId, playlistId, showArtwork)}
-  />
-);
+const EmbeddedAudio: React.FC<Props> = ({ tracks }) => {
+  const [isPlaying, setIsPlaying] = useState(false);
+  const [currentTime, setCurrentTime] = useState(0);
 
-export default EmbeddedAudio;
+  const currentTrack = useRef(0);
```

Comment:
why is current track a ref, and not state?

also, slight naming nitpick, could you rename it `currentTrackIndex`?